[SvelteKit 3]: There is no way to easily install a pre-release version of SvelteKit 3 with existing adapters via NPM

### Describe the bug

`package.json`
```
...
"@sveltejs/adapter-vercel": "7.0.0-next.1",
"@sveltejs/kit": "3.0.0-next.7",
...
```

The constraint `@sveltejs/kit@"^3.0.0"` does not include pre-release (`next`) versions. 

Is this a intentional restriction, or is it possible to expand the list of dependencies using a pre-release version of SvelteKit? (`^3.0.0-next.0` or, possibly, `>=3.0.0-next.0 <4`).

`npm install --legacy-peer-deps` solves the problem for local development, but it's inconvenient and complicates deployment.

I also believe that this applies only to NPM, and that there is no such problem with PNPM.

### Logs

```Shell
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @sveltejs/adapter-vercel@7.0.0-next.1
npm warn Found: @sveltejs/kit@3.0.0-next.7
npm warn node_modules/@sveltejs/kit
npm warn   dev @sveltejs/kit@"3.0.0-next.7" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peer @sveltejs/kit@"^3.0.0" from @sveltejs/adapter-vercel@7.0.0-next.1
npm warn node_modules/@sveltejs/adapter-vercel
npm warn   dev @sveltejs/adapter-vercel@"7.0.0-next.1" from the root project
npm warn ERESOLVE overriding peer dependency
npm warn While resolving: @sveltejs/adapter-vercel@7.0.0-next.1
npm warn Found: @sveltejs/kit@undefined
npm warn node_modules/@sveltejs/kit
npm warn   dev @sveltejs/kit@"3.0.0-next.7" from the root project
npm warn
npm warn Could not resolve dependency:
npm warn peer @sveltejs/kit@"^3.0.0" from @sveltejs/adapter-vercel@7.0.0-next.1
npm warn node_modules/@sveltejs/adapter-vercel
npm warn   dev @sveltejs/adapter-vercel@"7.0.0-next.1" from the root project
npm error code ETARGET
npm error notarget No matching version found for @sveltejs/kit@^3.0.0.
npm error notarget In most cases you or one of your dependencies are requesting a package version that doesn't exist.
```

### Severity

annoyance

--- a/.changeset/prerelease-peer-deps.md
+++ b/.changeset/prerelease-peer-deps.md
@@ -1,0 +1,9 @@
+---
+'@sveltejs/adapter-auto': patch
+'@sveltejs/adapter-cloudflare': patch
+'@sveltejs/adapter-node': patch
+'@sveltejs/adapter-static': patch
+'@sveltejs/adapter-vercel': patch
+---
+
+fix: allow prerelease versions of SvelteKit 3 to satisfy the peer dependency range

--- a/packages/adapter-auto/package.json
+++ b/packages/adapter-auto/package.json
@@ -46,6 +46,6 @@
 		"vitest": "catalog:"
 	},
 	"peerDependencies": {
-		"@sveltejs/kit": "^3.0.0"
+		"@sveltejs/kit": "^3.0.0-next.0"
 	}
 }

--- a/packages/adapter-cloudflare/package.json
+++ b/packages/adapter-cloudflare/package.json
@@ -56,7 +56,7 @@
 		"vitest": "catalog:"
 	},
 	"peerDependencies": {
-		"@sveltejs/kit": "^3.0.0",
+		"@sveltejs/kit": "^3.0.0-next.0",
 		"wrangler": "^4.67.0"
 	}
 }

--- a/packages/adapter-node/package.json
+++ b/packages/adapter-node/package.json
@@ -54,6 +54,6 @@
 		"rolldown": "^1.1.2"
 	},
 	"peerDependencies": {
-		"@sveltejs/kit": "^3.0.0"
+		"@sveltejs/kit": "^3.0.0-next.0"
 	}
 }

--- a/packages/adapter-static/package.json
+++ b/packages/adapter-static/package.json
@@ -48,6 +48,6 @@
 		"vite": "catalog:"
 	},
 	"peerDependencies": {
-		"@sveltejs/kit": "^3.0.0"
+		"@sveltejs/kit": "^3.0.0-next.0"
 	}
 }

--- a/packages/adapter-vercel/package.json
+++ b/packages/adapter-vercel/package.json
@@ -52,6 +52,6 @@
 		"vitest": "catalog:"
 	},
 	"peerDependencies": {
-		"@sveltejs/kit": "^3.0.0"
+		"@sveltejs/kit": "^3.0.0-next.0"
 	}
 }

